dashboard/client/src/pages/job/TaskTimeline.tsx

```diff
@@ -0,0 +1,231 @@
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  Button,
+  CircularProgress,
+  Grid,
+  List,
+  ListItem,
+  makeStyles,
+  Typography,
+} from "@material-ui/core";
+import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
+import dayjs from "dayjs";
+import React, { useState } from "react";
+import { AlertDialog } from "../../common/AlertDialog";
+import { DownloadTaskTimeline, getTaskTimeline } from "../../service/task";
+
+const PERFETTO_URL = "https://ui.perfetto.dev";
+const PERFETTO_LOADING_TIMEOUT_MS = 5000;
+
+const useStyle = makeStyles((theme) => ({
+  root: {
+    padding: theme.spacing(2),
+  },
+  taskProgressTable: {
+    marginTop: theme.spacing(2),
+  },
+}));
+
+// The module guarantees there's only one fetch at
+// a time meaning there can be only one timer at a time.
+let timer: null | any = undefined;
+
+export const TaskTimeline = ({ jobId = null }: { jobId: string | null }) => {
+  const classes = useStyle();
+
+  const TimelineDownloadButton = ({
+    jobId = null,
+  }: {
+    jobId: string | null;
+  }) => {
+    const [loading, setLoading] = useState(false);
+
+    const onClick = () => {
+      // Download the chrome tracing file.
+      if (!loading) {
+        setLoading(true);
+        // blob response type is necessary to
+        DownloadTaskTimeline(jobId)
+          .then((blob) => {
+            const url = window.URL.createObjectURL(blob.data);
```

Comment:
Not sure if it is the best way to support downloading files... It was from stackoverflow

Reply:
I adjusted this to be done in a more native way.